File: app/src/main/java/com/udacity/maluleque/meutako/model/Transaction.java
package com.udacity.maluleque.meutako.model;

import android.os.Parcel;
import android.os.Parcelable;

import com.google.firebase.firestore.IgnoreExtraProperties;

@IgnoreExtraProperties
public class Transaction implements Parcelable {
    private String uid;
    private String type;
    private double amount;
    private String category;
    private String description;
    private long date;
    private String image;

    public Transaction() {
    }

    public static final Creator<Transaction> CREATOR = new Creator<Transaction>() {
        @Override
        public Transaction createFromParcel(Parcel in) {
            return new Transaction(in);
        }

        @Override
        public Transaction[] newArray(int size) {
            return new Transaction[size];
        }
    };

    protected Transaction(Parcel in) {
        uid = in.readString();
        type = in.readString();
        amount = in.readDouble();
        category = in.readString();
        description = in.readString();
        date = in.readLong();
        image = in.readString();
    }

    @Override
    public void writeToParcel(Parcel dest, int flags) {
        dest.writeString(uid);
        dest.writeString(type);
        dest.writeDouble(amount);
        dest.writeString(category);
        dest.writeString(description);
        dest.writeLong(date);
        dest.writeString(image);
    }

    @Override
    public int describeContents() {
        return 0;
    }

    public String getUid() {
        return uid;
    }

    public void setUid(String uid) {
        this.uid = uid;
    }

    public String getType() {
        return type;
    }

    public void setType(String type) {
        this.type = type;
    }

    public double getAmount() {
        return amount;
    }

    public void setAmount(double amount) {
        this.amount = amount;
    }

    public String getCategory() {
        return category;
    }

    public void setCategory(String category) {
        this.category = category;
    }

    public String getDescription() {
        return description;
    }

    public void setDescription(String description) {
        this.description = description;
    }

    public long getDate() {
        return date;
    }

    public void setDate(long date) {
        this.date = date;
    }

    public String getImage() {
        return image;
    }

    public void setImage(String image) {
        this.image = image;
    }
}

File: app/src/main/java/com/udacity/maluleque/meutako/model/Transaction.kt
package com.udacity.maluleque.meutako.model

import android.os.Parcel
import android.os.Parcelable
import com.google.firebase.firestore.IgnoreExtraProperties
import com.udacity.maluleque.meutako.utils.DateUtils.getDataDayMonth

@IgnoreExtraProperties
class Transaction : Parcelable {
    var uid: String? = null
    var type: String? = null
    var amount = 0.0
    var category: String? = null
    var description: String? = null
    var date: Long = 0
    var image: String? = null

    constructor() {}

    protected constructor(`in`: Parcel) {
        uid = `in`.readString()
        type = `in`.readString()
        amount = `in`.readDouble()
        category = `in`.readString()
        description = `in`.readString()
        date = `in`.readLong()
        image = `in`.readString()
    }

    override fun writeToParcel(dest: Parcel, flags: Int) {
        dest.writeString(uid)
        dest.writeString(type)
        dest.writeDouble(amount)
        dest.writeString(category)
        dest.writeString(description)
        dest.writeLong(date)
        dest.writeString(image)
    }

    override fun describeContents(): Int {
        return 0
    }

    val formattedDate: String
        get() = getDataDayMonth(date)

    companion object {
        val CREATOR: Parcelable.Creator<Transaction?> = object : Parcelable.Creator<Transaction?> {
            override fun createFromParcel(`in`: Parcel): Transaction? {
                return Transaction(`in`)
            }

            override fun newArray(size: Int): Array<Transaction?> {
                return arrayOfNulls(size)
            }
        }
    }
}